### PR TITLE
Implement exit codes, handle no tests [fix #246]

### LINKED ISSUE #246: Stop after discover: 0 tests selected
Shouldn't be `0 tests selected` from 'discover' step a show stopper?

Further steps should cope with that though, if one wants to skip discovery and run plan just for provision/prepare.

Say I want to run test (`run tests --name`), it is quite reasonable that such test doesn't need to be in all plans. But pipeline continues and fails after provision and prepare. Those steps are time/resource consuming and that error is a false negative, IMHO.

In tmt repo, `tmt run -a tests --name /tests/discover/filtering prepare --how shell -s 'echo 1' report -vv`
```
/plans/basic
    discover
       summary: 1 test selected
 --- cut %< ---
      summary: 1 test passed

/plans/helps
    discover
        summary: 0 tests selected
 --- cut %< ---
    execute
        how: shell
    finish
        summary: 0 tasks completed
 ERROR  Test execution failed: ("Failed to run command './run.sh -v .. shell stdout.log stderr.log'.", './run.sh -v .. shell stdout.log stderr.log', '', '')
```

### PR DESCRIPTION
**Pull request by psss**
The basic set of exit codes documented and implemented.
Added total stats of the test results at the very end.
Moved results summary to the Result class for sharing.

### CODE REVIEW

**Comment by psss** [comment]
This should cover #246. @lukaszachy, please have a look.

File: tests/core/dry/test.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 
 rlJournalStart
     rlPhaseStartTest
-        rlRun "tmt run --dry"
-        rlRun "tmt run --dry -dv"
-        rlRun "tmt run --dry -ddvv"
-        rlRun "tmt run --dry -dddvvv"
+        rlRun "tmt run --dry" 3
```
Comment (lukaszachy) [inline tests/core/dry/test.sh:6]: I understand why it reports exit code 3 but I object. As a user I asked it to not run anything and tmt satisfied my requrest, but exitcode is non-zero which indicates error (usually)

Comment (psss) [reply tests/core/dry/test.sh]: Good point, adjusted in 7394537.

**Comment by psss** [comment]
Thanks for the review, @lukaszachy.

**Pull request merged**